### DISCUSSION TITLE
Is it possible to load delta_scan in the same way as iceberg?

### DISCUSSION THREAD

**Discussion by ns1000** [duckdb/pg_duckdb#199] [Q&A] [answered]
Is it possible to load the delta extension and run delta_scan in the same way as iceberg?

I tried the following:

CREATE OR REPLACE FUNCTION delta_scan(path text)
RETURNS SETOF record LANGUAGE 'plpgsql' AS
$func$
BEGIN
    RAISE EXCEPTION 'Function `delta_scan(TEXT)` only works with Duckdb execution.';
END;
$func$;

SELECT duckdb.install_extension('delta');

But this resulted in this error:

assets=# select * from delta_scan('/home/stats/datalake/lake/events/events') as clicks(id bigint, group_id bigint, group_type text, event_id bigint, event_type text, linked_object_id bigint, object_type text,time timestamp, created_at timestamp) limit 10;
WARNING:  (PGDuckDB/CreatePlan) Prepared query returned an error: 'Catalog Error: Table Function with name "delta_scan" is not in the catalog, but it exists in the delta extension.

Please try installing and loading the delta extension by running:
INSTALL delta;
LOAD delta;

Alternatively, consider enabling auto-install and auto-load by running:
SET autoinstall_known_extensions=1;
SET autoload_known_extensions=1;
ERROR:  Function `delta_scan(TEXT)` only works with Duckdb execution.
CONTEXT:  PL/pgSQL function delta_scan(text) line 3 at RAISE
assets=#

**Comment by ns1000** [comment]
Well i am kind of answering myself here, probably not the best solution but it works, i added this in the manager initalization:

pgduckdb_duckdb.cpp:        config.SetOptionByName("autoinstall_known_extensions", true);
pgduckdb_duckdb.cpp:        config.SetOptionByName("autoload_known_extensions", true);

This as suggested autoloads the plugin when it is encountered.

And i had to add the TIMESTAMP_TZ to the postgres conversion types.

**Comment by JelteF** [comment, marked as answer]
I'm pretty sure this was the same issue as described here: https://github.com/duckdb/pg_duckdb/issues/278

And that one is now fixed. So closing this. Native support for delta_scan has also been added in #403
